src/Cryptor.ts

```diff
@@ -1,131 +1,160 @@
-import * as fs from 'fs';
-import * as crypto from 'crypto';
-import * as process from 'process';
+import * as fs from 'fs'
+import * as crypto from 'crypto'
+import * as process from 'process'
+import { spawn, Worker, ModuleThread } from 'threads'
+import { CryptorWorker } from './CryptorWorker'
 
 // TODO: function docs
 
+// interface CryptorParameters {
+// 	key: Buffer | string
+// 	initVector?: Buffer
+// 	algorithm?: string
+// 	useWebWorkers?: boolean
+// }
+
 export default class Cryptor {
-  private _algo: string;
-  private readonly _initVector: Buffer;
-  private _key: Buffer;
-  private _cipher: crypto.Cipher;
-  private _decipher: crypto.Decipher;
-	constructor(pass: string, initVector: Buffer = null, algo: string = 'id-aes256-GCM') {
-		this._algo = algo;
-    	this._initVector = initVector ? initVector : this.genRandomIVSync();
+	private _algorithm: string
+	private _initVector: Buffer
+	private _key: Buffer
+	private _cipher: crypto.Cipher
+	private _decipher: crypto.Decipher
+	private _useWebWorkers: boolean
+	private _cryptorWorker?: ModuleThread<CryptorWorker>
+	constructor(
+		key: Buffer | string,
+		initVector: Buffer = crypto.randomBytes(16),
+		algorithm: string = 'aes-256-gcm',
+		useWebWorkers: boolean = false
+	) {
+		this._algorithm = algorithm
+		this._initVector = initVector
 		// TODO: generate salt ?
```

Comment:
For example here, if this is a real question, turn it into an issue.

Reply:
This has been turned into issue #22